src/gravel/controllers/nodes/mgr.py

```diff
@@ -493,63 +610,44 @@ def connmgr(self) -> ConnMgr:
     def token(self) -> Optional[str]:
         return self._token
 
-    def _get_node_file(self, what: str) -> Path:
-        confdir: Path = gstate.config.confdir
-        assert confdir.exists()
-        assert confdir.is_dir()
-        return confdir.joinpath(f"{what}.json")
+    @property
+    def store(self) -> KV:
+        assert self._kvstore
+        return self._kvstore
 
-    def _node_init(self) -> None:
-        statefile: Path = self._get_node_file("node")
-        if not statefile.exists():
-            # other control files must not exist either
-            tokenfile: Path = self._get_node_file("token")
-            assert not tokenfile.exists()
+    async def _obtain_state(self) -> None:
+        assert self._kvstore
 
-            state = NodeStateModel(
-                uuid=uuid4(),
-                role=NodeRoleEnum.NONE,
-                stage=NodeStageEnum.NONE,
-                address=None,
-                hostname=None
-            )
-            try:
-                statefile.write_text(state.json())
-            except Exception as e:
-                raise NodeError(str(e))
-            assert statefile.exists()
-
-        self._state = NodeStateModel.parse_file(statefile)
+        def _watcher(key: str, value: str) -> None:
+            if key == "/nodes/token":
+                logger.info(f"got updated token: {value}")
+                self._token = value
 
-    def _load(self) -> None:
-        self._token = self._load_token()
+        self._token = await self._load_token()
+        await self._kvstore.watch("/nodes/token", _watcher)
```

Comment:
Why is a token obtained, and then the key watched for updates? Do we expect the token to change?

Reply:
I hope so. In this case was meant mostly as a PoC that watching keys works, but thinking ahead, given I hope we'll let the user refresh their token should they want to.